src/models/behaviors.js

```diff
@@ -397,19 +547,94 @@ function copy (originalRequest, responsePromise, copyArray, logger) {
         copyArray.forEach(function (copyConfig) {
             var from = getFrom(originalRequest, copyConfig.from),
                 using = copyConfig.using || {},
-                fnMap = { regex: regexValue, xpath: xpathValue, jsonpath: jsonpathValue },
+                fnMap = {
+                    regex: regexValue,
+                    xpath: xpathValue,
+                    jsonpath: jsonpathValue
+                },
                 values = [];
 
             if (fnMap[using.method]) {
                 values = fnMap[using.method](from, copyConfig, logger);
             }
-
             replace(response, copyConfig.into, values, logger);
         });
         return Q(response);
     });
 }
 
+function CSVData (CSVPath, ColumnMatch, result, values, index) {
+    var flag = true;
+    var storeColumnIntoValues = [];
+    var CSVData0 = csvToObject({
+        filename: CSVPath
+    });
+    Object.keys(CSVData0).forEach(function (key) {
+        Object.keys(CSVData0[key]).forEach(function () {
+            var keyCheck = (CSVData0[key][ColumnMatch]);
+            if ((flag) && (defined(keyCheck)) && (keyCheck.localeCompare(values[index]) === 0)) {
+                for (var t = 1; t <= result.length; t += 1) {
+                    var intoSubset = result[t - 1];
+                    var outputCheck = CSVData0[key][intoSubset];
+                    if (defined(outputCheck)) {
+                        storeColumnIntoValues.push(outputCheck.trim());
+                    }
+                    flag = false;
+                }
+            }
+        });
+    });
+    return storeColumnIntoValues;
+}
+
+function CSVColumnIntoValue (obj) {
+    var result = [];
+    Object.keys(obj).forEach(function (key) {
+        result.push(obj[key]);
+    });
+    return result;
+}
+
+
+function lookup (originalRequest, responsePromise, csvobject, logger) {
```

Comment:
Maybe name csvobject `lookupArray`. It took me a minute before I realized it was the array instead of one of the objects, and at this point it's not CSV-specific (even though that's only one implemented yet).